modules/market/src/main/java/com/opengamma/strata/market/curve/SeasonalNodalCurve.java

```diff
@@ -0,0 +1,638 @@
+   * @param lastFixingValue  the value of the last fixing
+   * @param seasonalityDefinition  the seasonality definition, which is made of month-on-month adjustment 
+   * and the adjustment type
+   * @return the seasonal curve instance
+   */
+  public static SeasonalNodalCurve of(
+      NodalCurve curveWithoutFixing, 
+      LocalDate valuationDate,
+      YearMonth lastMonth,
+      double lastFixingValue,
+      SeasonalityDefinition seasonalityDefinition) {
+
+    ArgChecker.isTrue(lastMonth.isBefore(YearMonth.from(valuationDate)), 
+        "Last fixing month must be before valuation date");
+    double nbMonth = YearMonth.from(valuationDate).until(lastMonth, MONTHS);
+    DoubleArray x = curveWithoutFixing.getXValues();
+    ArgChecker.isTrue(nbMonth < x.get(0), "The first estimation month should be after the last known index fixing");
+    NodalCurve extendedCurve = curveWithoutFixing.withNode(nbMonth, lastFixingValue, ParameterMetadata.empty());
+    double[] seasonalityCompoundedArray = new double[12];
+    int lastMonthIndex = lastMonth.getMonth().getValue() - 2;
+    seasonalityCompoundedArray[(int) ((nbMonth + 12 + 1) % 12)] = 
+        seasonalityDefinition.getSeasonalityMonthOnMonth().get((lastMonthIndex + 1) % 12);
+    for (int i = 1; i < 12; i++) {
+      int j = (int) ((nbMonth + 12 + 1 + i) % 12);
+      seasonalityCompoundedArray[j] = seasonalityDefinition.getAdjustmentType()
+          .applyShift(seasonalityCompoundedArray[(j - 1 + 12) % 12], 
+              seasonalityDefinition.getSeasonalityMonthOnMonth().get((lastMonthIndex + 1 + i) % 12));
+    }
+    return new SeasonalNodalCurve(extendedCurve, DoubleArray.ofUnsafe(seasonalityCompoundedArray), 
+        seasonalityDefinition.getAdjustmentType());
+  }
+
+  //-------------------------------------------------------------------------
+  // restricted constructor
+  @ImmutableConstructor
+  private SeasonalNodalCurve(
+      NodalCurve curve, 
+      DoubleArray seasonality, 
+      ShiftType adjustmentType) {
+    this.curve = curve;
+    this.seasonality = seasonality;
+    this.xFixing = curve.getXValues().get(0);
+    this.yFixing = curve.getYValues().get(0);
+    int i = seasonalityIndex(xFixing);
+    ArgChecker.isTrue(adjustmentType.applyShift(yFixing, seasonality.get(i) ) - yFixing < 1.0E-10, 
+        "Fixing value should be unadjusted");
+    this.adjustmentType = adjustmentType;
+  }
+
+  @Override
+  public CurveMetadata getMetadata() {
+    return curve.getMetadata();
+  }
+
+  @Override
+  public double yValue(double x) {
+    int i = seasonalityIndex(x);
+    double adjustment = seasonality.get(i);
+    return adjustmentType.applyShift(curve.yValue(x), adjustment);
+  }
+  
+  // The index on the seasonality vector associated to a time (nb months).
+  private int seasonalityIndex(double x) {
+    long xLong = Math.round(x);
+    return (int) ((xLong + 12) % 12); // Shift by 12 has java compute the remainder of negative numbers as negative
```

Comment:
Do you mean `(((xLong % 12) + 12) % 12)` to always get a positive number? The code above can overflow Integer.MAX and can return a negative number if xLong < 12.